src/cart/crt_context.c

```diff
@@ -902,6 +1037,15 @@ crt_req_timeout_hdlr(struct crt_rpc_priv *rpc_priv)
 		d_tm_inc_counter(crt_ctx->cc_timedout, 1);
 
 	switch (rpc_priv->crp_state) {
+	case RPC_STATE_QUEUED:
+		RPC_ERROR(rpc_priv, "aborting waiting to group %s, rank %d, tgt_uri %s\n",
+			  grp_priv->gp_pub.cg_grpid, tgt_ep->ep_rank, rpc_priv->crp_tgt_uri);
+		D_MUTEX_LOCK(&rpc_priv->crp_epi->epi_mutex);
+		d_list_del_init(&rpc_priv->crp_epi_link);
```

Comment:
I think you're right that we need to RPC_DECREF here. I've modified crt_context_req_untrack a little (otherwise it can't handle QUEUED request) and used it here.